lib/matplotlib/tests/test_axes.py

```diff
@@ -105,6 +105,30 @@ def test_label_loc_rc(fig_test, fig_ref):
     cbar.set_label("Z Label", x=1, ha='right')
 
 
+def test_label_shift():
+    fig, ax = plt.subplots()
+
+    # Test label re-centering on x-axis
+    ax.set_xlabel("Test label", loc="left")
+    ax.set_xlabel("Test label", loc="center")
+    assert ax.xaxis.get_label()._x == 0.5
```

Comment:
We try to only test public interface and `._x` is private.  Can you change to a test of the public getter?

Reply:
@jklymak Ah my bad; thanks for the feedback! Updated the code. Let me know if it looks good now.